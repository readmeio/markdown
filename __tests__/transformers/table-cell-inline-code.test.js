import { hast, md, mdast } from '../../index';

describe('tableCellInlineCode', () => {
  it('unescapes escaped pipe chars inside inline code within table headers', () => {
    const doc = `
| \`one \\| two \\| three \\| four\` | two |
| :- | :- |
`;

    const tree = hast(doc);
    expect(tree).toMatchSnapshot();
  });

  it('unescapes escaped pipe chars inside inline code within table cells', () => {
    const doc = `
|    |    |
| :- | :- |
| \`one \\| two \\| three \\| four\` | two |
`;

    const tree = hast(doc);
    expect(tree).toMatchSnapshot();
  });

  it('preserves escaped pipe chars inside text table cells', () => {
    const doc = `
| these \\| stay \\| escaped \\| inside \\| a single cell | |
| :- | :- |
`;

    const tree = hast(doc);
    expect(tree).toMatchSnapshot();
  });

  it('splits table cells when inline code contains "unescaped" pipe chars', () => {
    const doc = `
| \`this | splits | up | to | more | cells\` | two |
| :- | :- |
`;

    const tree = hast(doc);
    expect(tree).toMatchSnapshot();
  });

  it('preserves the escaped pipe character when re-serializing from mdast', () => {
    const doc = `
| \`one \\| two \\| three \\| four\` | two |
| :- | :- |
`;

    const tree = mdast(doc);
    expect(md(tree)).toMatchInlineSnapshot(`
      "| \`one \\\\| two \\\\| three \\\\| four\` | two |
      | :---------------------------- | :-- |
      "
    `);
  });
});
