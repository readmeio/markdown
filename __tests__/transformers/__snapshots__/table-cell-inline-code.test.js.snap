// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableCellInlineCode preserves escaped pipe chars inside text table cells 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "






",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "these | stay | escaped | inside | a single cell",
                    },
                  ],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
                Object {
                  "children": Array [],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
              ],
              "properties": Object {},
              "tagName": "tr",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "thead",
          "type": "element",
          "value": "these | stay | escaped | inside | a single cell",
        },
      ],
      "properties": Object {},
      "tagName": "table",
      "type": "element",
    },
  ],
  "data": Object {
    "quirksMode": false,
  },
  "type": "root",
}
`;

exports[`tableCellInlineCode splits table cells when inline code contains "unescaped" pipe chars 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "






",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "\`this",
                    },
                  ],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "splits",
                    },
                  ],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
              ],
              "properties": Object {},
              "tagName": "tr",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "thead",
          "type": "element",
          "value": "\`this splits",
        },
      ],
      "properties": Object {},
      "tagName": "table",
      "type": "element",
    },
  ],
  "data": Object {
    "quirksMode": false,
  },
  "type": "root",
}
`;

exports[`tableCellInlineCode unescapes escaped pipe chars inside inline code within table cells 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "






",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "one | two | three | four",
                        },
                      ],
                      "properties": Object {},
                      "tagName": "code",
                      "type": "element",
                    },
                  ],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "two",
                    },
                  ],
                  "properties": Object {
                    "align": "left",
                  },
                  "tagName": "th",
                  "type": "element",
                },
              ],
              "properties": Object {},
              "tagName": "tr",
              "type": "element",
            },
          ],
          "properties": Object {},
          "tagName": "thead",
          "type": "element",
          "value": "one | two | three | four two",
        },
      ],
      "properties": Object {},
      "tagName": "table",
      "type": "element",
    },
  ],
  "data": Object {
    "quirksMode": false,
  },
  "type": "root",
}
`;
